File: frontend/projects/admin/src/app/dto/mouvement-stock.dto.ts
export interface MouvementStockDTO {
  id?: number;
  produitId: number;
  /** Offre concernée : le stock est porté par l'offre, pas par le produit. */
  offerId?: number;
  /** Réservation à l'origine du mouvement, si elle existe. */
  reservationId?: number;
  typeMouvement: string; // ENTREE, SORTIE, RESERVATION, LIBERATION_RESERVATION, VENTE, AJUSTEMENT, INVENTAIRE
  quantite: number;
  /** Stock disponible de l'offre après le mouvement. */
  quantiteApresMouvement: number;
  reference: string;
  referenceType: string; // DEMANDE, COMMANDE, INVENTAIRE, AJUSTEMENT
  motif: string;
  isActive?: boolean;
  createdAt?: string; // ISO date string
  updatedAt?: string; // ISO date string
  createdBy?: number;
}

File: backend/src/main/java/com/vollailelink/backend/dto/MouvementStockDTO.java
package com.vollailelink.backend.dto;

import com.vollailelink.backend.model.enums.ReferenceTypeStock;
import com.vollailelink.backend.model.enums.TypeMouvementStock;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MouvementStockDTO {
    private Long id;
    private Long produitId;
    private String produitNom;
    /**
     * Offre concernee : le stock est porte par l'offre, pas par le produit.
     * Sans ce champ, impossible de savoir quel lot a bouge.
     */
    private Long offerId;
    /** Reservation a l'origine du mouvement, si elle existe. */
    private Long reservationId;
    private TypeMouvementStock typeMouvement;
    private Integer quantite;
    private Integer quantiteApresMouvement;
    private String reference;
    private ReferenceTypeStock referenceType;
    private String motif;
    private Boolean isActive;
    private OffsetDateTime createdAt;
}
